web/packages/teleport/src/Discover/Server/ConnectAwsAccount/ConnectAwsAccount.tsx

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import {
+  Box,
+  ButtonText,
+  Text,
+  ButtonPrimary,
+  Indicator,
+  Alert,
+  Flex,
+} from 'design';
+import FieldSelect from 'shared/components/FieldSelect';
+import useAttempt from 'shared/hooks/useAttemptNext';
+import { Option as BaseOption } from 'shared/components/Select';
+import Validation, { Validator } from 'shared/components/Validation';
+import { requiredField } from 'shared/components/Validation/rules';
+import TextEditor from 'shared/components/TextEditor';
+
+import cfg from 'teleport/config';
+import {
+  Integration,
+  IntegrationKind,
+  integrationService,
+} from 'teleport/services/integrations';
+import { integrationRWEAndNodeRWE } from 'teleport/Discover/yamlTemplates';
+import useTeleport from 'teleport/useTeleport';
+
+import { ActionButtons, HeaderSubtitle, Header } from '../../Shared';
+
+import {
+  DiscoverUrlLocationState,
+  NodeMeta,
+  useDiscover,
+} from '../../useDiscover';
+
+type Option = BaseOption<Integration>;
+
+export function ConnectAwsAccount() {
+  const { storeUser } = useTeleport();
+  const {
+    prevStep,
+    nextStep,
+    agentMeta,
+    updateAgentMeta,
+    eventState,
+    resourceSpec,
+    currentStep,
+  } = useDiscover();
+
+  const integrationAccess = storeUser.getIntegrationsAccess();
+  const nodesAccess = storeUser.getNodeAccess();
+  const hasAccess =
+    integrationAccess.create &&
+    integrationAccess.list &&
+    // Required access after integrating:
+    integrationAccess.use &&
+    nodesAccess.create &&
+    nodesAccess.edit &&
+    nodesAccess.list &&
+    nodesAccess.read;
```

Comment:
I added this because I thought it would be needed for the `TestConnection` step, though I guess we could remove it since technically you should still be able to create the node and EICE without it. What do you think?